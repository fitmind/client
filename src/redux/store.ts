import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
