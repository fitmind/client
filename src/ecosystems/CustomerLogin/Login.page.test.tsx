import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './Login.page';
import { Router } from 'react-router';
import store, { history } from '../../redux/store';
import { Provider } from 'react-redux';

describe('LoginPage', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    // @ts-ignore
                    <LoginPage />
                </Router>
            </Provider>,
        );
    });
});
