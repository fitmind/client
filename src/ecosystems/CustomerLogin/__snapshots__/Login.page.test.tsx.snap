// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col"
    >
      <div
        className="sc-bdVaJa fOsasd"
      >
        <div
          className="card"
        >
          <h5
            className="card-header"
          >
            Login
          </h5>
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              Please enter with your admin login credentials
            </p>
            <form
              className=""
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter email"
                  type="email"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="formBasicPassword"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="formBasicPassword"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
                <small
                  className="text-muted form-text"
                >
                  Password needs to have numbers and letters
                </small>
              </div>
              <button
                className="btn btn-outline-primary btn-block"
                disabled={false}
                type="submit"
              >
                Login
              </button>
              <button
                className="btn btn-outline-secondary btn-block"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Register
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</div>
`;
